### PR TITLE
Array format placement

### PR DESCRIPTION
**Pull request by jessemyers**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #1178

## What is the new behavior?

Fixes the bug.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

The set of possible fields that should be moved from the outer type to the inner type in this case could be expanded. I picked a set of fields that match my own use cases, but these might not be exhaustive!

### CODE REVIEW

File: lib/services/schema-object-factory.ts [hunk 1]
```diff
@@ -424,15 +425,20 @@ export class SchemaObjectFactory {
     type: string | Record<string, any>
   ): SchemaObjectMetadata {
     const keysToRemove = ['type', 'enum'];
+    const keysToMove = [
+      'format',
```
Comment (jessemyers) [inline lib/services/schema-object-factory.ts:429]: The original bug report was only for `format`, but these other keys are all similarly applicable to the inner type and are fields that my own projects use. There may be other fields that belong here.

File: test/services/fixtures/create-user.dto.ts [hunk 2]
```diff
@@ -46,7 +46,8 @@ export class CreateUserDto {
 
   @ApiProperty({
     type: String,
-    isArray: true
+    isArray: true,
+    format: 'uri'
```
Comment (jessemyers) [inline test/services/fixtures/create-user.dto.ts:50]: This fixture did not previously define a `format`, but it looked like an appropriate place to demonstrate the expected behavior.

**Review by adrian-hippo — APPROVED** [review]

**Comment by jessemyers** [comment]
Hey @kamilmysliwiec, are you expecting anything else here?

**Comment by kamilmysliwiec** [comment]
LGTM

**Pull request merged**